packages/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts

```diff
@@ -214,6 +205,12 @@ export const useGridRowSelection = (
         return false;
       }
 
+      // If `keepNonExistentRowsSelected` is true, we might run in a case where row selectability is checked for a row that does not exist.
+      // Since that row was previously selected (otherwise it would not be checked at this point), we return true.
```

Comment:
> otherwise it would not be checked at this point

Are you sure about this?
For example `selectRow()` calls `isRowSelectable()` both with the `isSelected` argument values `true` and `false`.

Would it be better to return `true` if the row was previously selected but not in the lookup now?